File: programmers/python 3/42862.py
# 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42862

def solution(n, lost, reserve):
    reserve_lost = set(lost).intersection(set(reserve))
    lost = sorted(set(lost) - reserve_lost)
    reserve = list(set(reserve) - reserve_lost)

    for p in list(lost):
        if p - 1 in reserve:
            reserve.remove(p - 1)
            lost.remove(p)
        elif p + 1 in reserve:
            reserve.remove(p + 1)
            lost.remove(p)
    return n - len(lost)

File: programmers/swift/42862.swift
// 문제 링크: https://programmers.co.kr/learn/courses/30/lessons/42862

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    let reserveLost = Set(lost).intersection(Set(reserve))
    var lost = Set(lost).subtracting(reserveLost).sorted()
    var reserve = Array(Set(reserve).subtracting(reserveLost))

    Array(lost).forEach { person in
        guard let index = lost.firstIndex(of: person) else { return }

        if let prevIndex = reserve.firstIndex(of: person - 1) {
            reserve.remove(at: prevIndex)
            lost.remove(at: index)
        } else if let nextIndex = reserve.firstIndex(of: person + 1) {
            reserve.remove(at: nextIndex)
            lost.remove(at: index)
        }
    }

    return n - lost.count
}